app/waf/src/controllers/application.controller.ts

```diff
@@ -250,6 +259,37 @@ export class ApplicationController {
       }
     }
 
+    let rules = [];
+    let wafs = [];
+    if (service) {
+      params.service = service;
+      let endpointpolicyId = service.endpointpolicy;
+      if (endpointpolicyId !== undefined ) {
+        let endpointpolicy = await this.endpointpolicyRepository.findById(
+          endpointpolicyId,
+        );
+        if (endpointpolicy) {
+          params.endpointpolicy = endpointpolicy;
+          if (endpointpolicy.rules !== undefined) {
+            for (let ruleId of endpointpolicy.rules) {
+              let rule = await this.ruleRepository.findById(ruleId);
```

Comment:
How about to use `ruleRepository.find` to remove the for loop, what's more no exception.